### PR TITLE
SC-6019 - Having a tabbar for showing classes

### PR DESCRIPTION
**Pull request by DanielRieckhof**
# Description
The User wants to see the classes separated by years to get a better overview.

The updated view provides 3 tabs
  - Upcoming: All upcoming classes
  - Current: The classes of the current school year
  - Archive: The rest of classes

The default tab is the current school year. 
The tab "Archive" allows to filter within the list by school years. The default filter is the last school year.

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-6019

https://github.com/hpi-schul-cloud/schulcloud-client/pull/2096
(this includes the CSS update for the text color within inactive tabs)

## Changes
  - Used the Tabbar component for tabs
  - Limit years filter depending on active tab
  - Updated feather-filters query injection
  - Updated Tabbar handling to handle tabs with links (no content switch on page)

It might be useful to check other usages of the tabbar, e.g. the curse overview.

## Screenshots of UI changes

There is a known issue with the text color within the inactive tab. The required CSS update has been done here:
https://github.com/hpi-schul-cloud/schulcloud-client/pull/2096

image

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=6037847&bid=19785162)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=6037847&bid=19785162)

File: static/scripts/administration.js [hunk 1]
```diff
@@ -15,10 +15,17 @@ window.addEventListener('DOMContentLoaded', () => {
...
+
 			const newurl = `?filterQuery=${
 				escape(JSON.stringify(filter[0]))
-			}&p=${
-				getQueryParameterByName('p')}`;
+			}${
+				page ? `&p=${page}` : ''
+			}${
+				showTab ? `&showTab=${showTab}` : ''
+			}`;
```
Comment (CeEv) [inline static/scripts/administration.js:28]: let page = getQueryParameterByName('p');
page = page ? `&p=${page}` : '';

let showTab = getQueryParameterByName('showTab');
showTab = showTab ? `&showTab=${showTab}` : '';

const filterQuery =`?filterQuery=${escape(JSON.stringify(filter[0]))};

const newurl = `${ filterQuery }${ page }${ showTab  };

**Comment by ABialoskorski** [comment]
please add a screenshot how it looks right now ;)

File: static/scripts/administration.js (hunk 1 above)
Comment (DanielRieckhof) [reply static/scripts/administration.js]: You're absolutely right. This makes the code more readable. Thank you for the input.

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=6037847&bid=19785162)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=6037847&bid=19785162)

**Comment by tofixx** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-client/pullRequest?prid=6045502&bid=19807385)

**Comment by tofixx** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-client/pullRequest?prid=6045502&bid=19807385)

**Comment by tofixx** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-client/pullRequest?prid=6045502&bid=19807385)

**Comment by tofixx** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-client/pullRequest?prid=6045502&bid=19807385)

**Comment by tofixx** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- static/scripts/tabbar.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-client/pullRequest?prid=6045502&bid=19807385)

**Review by tofixx — APPROVED** [review]

**Review by monaweitzenberg — APPROVED** [review]

**Pull request merged**